### PR TITLE
A task to diff locks files.

### PR DESCRIPTION
**Pull request by rspieldenner**
…ain plugin.

### CODE REVIEW

File: src/test/groovy/nebula/plugin/dependencylock/tasks/DiffLockTaskSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+            it.newLockFile = newLock
+        }
+
+        when:
+        def realizedTask = task.get()
+        realizedTask.diffLocks()
+
+        then:
+        realizedTask.diffFile.text == '''\
+            inconsistent:
```
Comment (OdysseusLives) [inline src/test/groovy/nebula/plugin/dependencylock/tasks/DiffLockTaskSpec.groovy]: This `inconsistent` section looks good!

Comment (OdysseusLives) [inline src/test/groovy/nebula/plugin/dependencylock/tasks/DiffLockTaskSpec.groovy]: This inconsistent section looks good!

**Comment by rahulsom** [comment]
I feel less confident about refactoring `Map<String, Map<String, String>>`. If this code is expected to live long enough, I would appreciate it being 
```
class LockFile {
  Map<String,Configuration> configurations
}
class Configuration {
  Map<String, Dependency> dependencies
}
class Dependency {
  String locked
  String requested
  ...
}
```

File: src/main/groovy/nebula/plugin/dependencylock/tasks/DiffLockTask.groovy [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            versionConfigurations.forEach { versions, configurations ->
+                collector.add("  ${versions.getFirst()} -> ${versions.getSecond()} [${configurations.sort().join(',')}]")
+            }
+            collector
+        }
+
+        return ['new': newDependencies, 'removed': removedDependencies, 'updated': updatedDependencies, 'inconsistent': inconsistentDependencies]
+    }
+
+    Map<String, SortedMap<String, Tuple2<String, String>>> collectConfigurationDiffs(Map<String, Map<String, String>> old, Map<String, Map<String, String>> updated) {
```
Comment (chali) [inline src/main/groovy/nebula/plugin/dependencylock/tasks/DiffLockTask.groovy]: Do you think we should use some explicit types?

Comment (rspieldenner) [reply src/main/groovy/nebula/plugin/dependencylock/tasks/DiffLockTask.groovy]: some types added, and methods simplified

File: src/main/groovy/nebula/plugin/dependencylock/DependencyLockTaskConfigurer.groovy [hunk 3]
```diff
@@ -306,4 +312,19 @@ class DependencyLockTaskConfigurer {
             false
         }
     }
+
+    private TaskProvider<DiffLockTask> configureDiffLockTask(String lockFileName, DependencyLockExtension extension) {
+        TaskProvider<DiffLockTask> diffLockTask = project.tasks.register(DIFF_LOCK_TASK_NAME, DiffLockTask) { diffTask ->
```
Comment (chali) [inline src/main/groovy/nebula/plugin/dependencylock/DependencyLockTaskConfigurer.groovy]: Nice! lazy creation.

**Pull request merged**